tools/webpack-make.js

```diff
@@ -83,3 +64,29 @@ function process_result(err, stats) {
             process.exit(1);
     }
 }
+
+async function build() {
+    // dynamic imports which need node_modules
+    const config = (await import(config_path)).default;
+    const webpack = (await import('webpack')).default;
+    const cockpit_rsync = (await import('../pkg/lib/cockpit-rsync-plugin.js'));
+
+    if (args.rsync) {
+        process.env.RSYNC = args.rsync;
+        config.plugins.push(new cockpit_rsync.CockpitRsyncWebpackPlugin({ source: "dist/" + (args.onlydir || "") }));
+    }
+
+    const compiler = webpack(config);
+
+    if (args.watch) {
+        compiler.hooks.watchRun.tap("WebpackInfo", compilation => {
+            const time = new Date().toTimeString().split(' ')[0];
+            process.stdout.write(`${time} Build started\n`);
+        });
+        compiler.watch(config.watchOptions, process_result);
+    } else {
+        compiler.run(process_result);
+    }
+}
+
+build();
```

Comment:
No `await` or `.run()` or similar?  How does it know when to quit?

Reply:
Top-level await is always a bit problematic. Having an `async main()` and calling it (without await) is the suggested approach.